content/posts/0025-oracle-graalvm-support.md

```diff
@@ -0,0 +1,87 @@
+    $BPL_JFR_ENABLED             false                                                        enables Java Flight Recording (JFR)
+    $BPL_JMX_ENABLED             false                                                        enables Java Management Extensions (JMX)
+    $BPL_JMX_PORT                5000                                                         configure the JMX port
+    $BPL_JVM_HEAD_ROOM           0                                                            the headroom in memory calculation
+    $BPL_JVM_LOADED_CLASS_COUNT  35% of classes                                               the number of loaded classes in memory calculation
+    $BPL_JVM_THREAD_COUNT        250                                                          the number of threads in memory calculation
+    $JAVA_TOOL_OPTIONS                                                                        the JVM launch flags
+    Using buildpack default Java version 17
+  Oracle GraalVM 17.0.9: Contributing to layer
+    Downloading from https://download.oracle.com/graalvm/17/archive/graalvm-jdk-17.0.9_linux-x64_bin.tar.gz
+    Verifying checksum
+    Expanding to /layers/paketo-buildpacks_oracle/native-image-svm
+    Adding 137 container CA certificates to JVM truststore
+    Writing env.build/JAVA_HOME.override
+    Writing env.build/JDK_HOME.override
+...
+```
+
+This example shows one of my favorite features of buildpacks, which is that switching JVM and Native Image vendors is trivial. The Oracle GraalVM is fantastic, but if you want to switch to use the GraalVM Community Edition, that's easy too. Just swap in `paketo-buildpacks/graalvm` for `paketo-buildpacks/oracle`. A two-second change and you're building using a different vendor.
+
+Just as easily, you can swap versions too. The Oracle buildpack presently supports both Java 17 and 21, which you can select by setting the `BP_JVM_VERSION` flag. For example, `pack build applications/native-image -b paketobuildpacks/oracle -b urn:cnb:builder:paketo-buildpacks/java-native-image --builder paketobuildpacks/builder-jammy-tiny --env BP_MAVEN_ACTIVE_PROFILES=native --env BP_JVM_VERSION=21`. This makes trying out new versions trivial, which ultimately makes your upgrades easier.
+
+### Spring Boot Build Tools
+
+If you're using Spring Boot Build Tools to generate your application image, you can use Oracle GraalVM with the following changes.
+
+1. Modify the image configuration block in your `pom.xml`.
+
+    ```
+    ...
+    <configuration>
+        <image>
+            <builder>paketobuildpacks/builder-jammy-tiny:latest</builder>
+            <buildpacks>
+                <buildpack>docker.io/paketobuildpacks/oracle</buildpack>
+                <buildpack>urn:cnb:builder:paketo-buildpacks/java-native-image</buildpack>
+            </buildpacks>
+        </image>
+    </configuration>
+    ...
+    ```
+
+2. Then run `./mvnw spring-boot:build-image -Pnative` to build your image.
+
```

Comment:
```suggestion

And the equivalent change with Gradle `build.gradle`:
    ```
    ...
    tasks.named("bootBuildImage") {
        environment = ['BP_NATIVE_IMAGE': 'true']
        builder = "paketobuildpacks/builder-jammy-tiny:latest"
        buildpacks = ["docker.io/paketobuildpacks/oracle", "urn:cnb:builder:paketo-buildpacks/java-native-image"]
    }
    ...
    ```
2. Then run `./gradlew bootBuildImage` to build your image.

```

Reply:
it was already set in https://github.com/paketo-buildpacks/samples/blob/main/java/native-image/spring-boot-native-image-gradle/build.gradle#L18 ; maybe it's no longer necessary...

Reply:
so I retried with this env. variable, and it worked fine.
Thanks for spotting this simplication!